js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.vertical.js

```diff
@@ -58,8 +58,11 @@ class VerticalRenderingStrategy extends BaseAppointmentsStrategy {
     _getItemPosition(item) {
         const allDay = this.isAllDay(item);
         const isRecurring = !!this.instance.fire('getField', 'recurrenceRule', item);
-        const isAppointmentLong = this.instance.fire('appointmentTakesSeveralDays', item);
-        const appointmentStartDate = this.startDate(item, true); // NOTE: Timezones and timestamp
+        const appointmentStartDate = this.startDate(item, true);
+        const appointmentEndDate = this.endDate(item);
+
+        // NOTE: subtract 1 millisecond to avoid 00.00 time
+        const isAppointmentTakesSeveralDays = !dateUtils.sameDate(appointmentStartDate, new Date(appointmentEndDate.getTime() - 1));
```

Comment:
This code looks strange, and if we really need it, it should be as an utility method or as a helper method of the appointment instance.